tests/cstor/gtest/test_zrepl_prot.cc

```diff
@@ -2020,7 +2045,31 @@ TEST(Snapshot, CreateAndDestroy) {
 	EXPECT_EQ(hdr_in.version, REPLICA_VERSION);
 	EXPECT_EQ(hdr_in.opcode, ZVOL_OPCODE_SNAP_LIST);
 	EXPECT_EQ(hdr_in.status, ZVOL_OP_STATUS_OK);
-	EXPECT_EQ(hdr_in.io_seq, 3);
+	EXPECT_EQ(hdr_in.io_seq, io_seq);
+	ASSERT_GE(hdr_in.len, sizeof (struct zvol_snapshot_list));
+	buf = (char *)malloc(hdr_in.len);
+	rc = read(control_fd, buf, hdr_in.len);
+	ASSERT_EQ(rc, hdr_in.len);
+	snaplist = (struct zvol_snapshot_list *) buf;
+	output = execCmd("zfs", std::string("get guid -Hpo value ") +
+	    vol_name);
+	EXPECT_EQ(snaplist->zvol_guid, std::stoul(output));
+	verify_snapshot_details(vol_name, snaplist->data, "");
+
+	// Try to fetch snapshot list using snapshot name
```

Comment:
add one more test to send wrong name in the snaplist command.

Reply:
done